enginetest/server_engine.go

```diff
@@ -213,7 +284,12 @@ func convertValue(sch sql.Schema, row sql.Row) sql.Row {
 			}
 		case query.Type_JSON:
 			if row[i] != nil {
-				row[i] = types.MustJSON(string(row[i].([]byte)))
+				// TODO: dolt returns the json result without escaped quotes and backslashes, which does not Unmarshall
+				r, err := attemptUnmarshalJSON(string(row[i].([]byte)))
+				if err != nil {
+					// we should use `row[i] = types.MustJSON()`
```

Comment:
Maybe consider skipping the test if we hit this? (Although, I see the `t.Testing` isn't wired in here yet). I'm just thinking about how we can get more visibility out of how well the tests are running. Skipped tests seem like they could be a helpful tool for us as we add more coverage with the server tests.